pkg/gofr/datasource/solr/solr_test.go

```diff
@@ -0,0 +1,178 @@
+package solr
+
+import (
+	"bytes"
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+)
+
+func Test_InvalidRequest(t *testing.T) {
+	_, err := call(context.TODO(), "GET", ":/localhost:", nil, nil)
```

Comment:
```suggestion
	_, err := call(context.Background()), "GET", ":/localhost:", nil, nil)
```

There is no reason to use a TODO context here (and everywhere in the tests). TODO is there when adding a context somewhere there was no context, but they might be a context at an higher level in the stack, so the TODO is there to say, there are things to do.

Here you are writing a test